server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -317,18 +325,49 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         }
     }
 
-    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+    public void acquireLock(long primaryTerm, long generation, String resourceID) throws IOException {
```

Comment:
I am not sure if directory should be aware of the locking.

Reply:
before acquiring the lock, we are making sure that a metadata file corresponding to a given commit is present in Remote Store.